lambdas/data-migration2/tests/test-granulesAndFiles.js

```diff
@@ -900,11 +900,343 @@ test.serial('migrateGranuleAndFilesViaTransaction processes granule with no file
+    type: 'data',
+    source: 'source4',
+  });
+
+  // change api (dynamo) granule files
+  testGranule.files = [testFile1, fakeFile3, fakeFile4];
+
+  // migrate 2nd time
+  const secondResult = await migrateGranuleAndFilesViaTransaction({
+    dynamoRecord: testGranule,
+    knex,
+    loggingInterval: 1,
+    migrateAndOverwrite: 'true',
+  });
+
+  const records = await t.context.granulePgModel.search(t.context.knex, {});
+  const fileRecords = await t.context.filePgModel.search(t.context.knex, {});
+  t.is(records.length, 1);
+  t.is(fileRecords.length, 3);
+
+  t.deepEqual(secondResult, {
+    filesResult: {
+      total_dynamo_db_records: 3,
+      failed: 0,
+      skipped: 0,
+      migrated: 3,
+    },
+    granulesResult: {
+      total_dynamo_db_records: 1,
+      failed: 0,
+      skipped: 0,
+      migrated: 1,
+    },
+  });
+
+  t.deepEqual(
+    omit(fileRecords[0], fileOmitList),
+    {
+      bucket: testFile1.bucket,
+      checksum_value: testFile1.checksum,
+      checksum_type: testFile1.checksumType,
+      key: testFile1.key,
+      path: null,
+      file_size: testFile1.size.toString(),
+      file_name: testFile1.fileName,
+      source: testFile1.source,
+      type: testFile1.type,
+    }
+  );
+  t.deepEqual(
+    omit(fileRecords[1], fileOmitList),
+    {
+      bucket: fakeFile3.bucket,
+      checksum_value: fakeFile3.checksum,
+      checksum_type: fakeFile3.checksumType,
+      key: fakeFile3.key,
+      path: null,
+      file_size: fakeFile3.size.toString(),
+      file_name: fakeFile3.fileName,
+      source: fakeFile3.source,
+      type: fakeFile3.type,
+    }
+  );
+  t.deepEqual(
+    omit(fileRecords[2], fileOmitList),
+    {
+      bucket: fakeFile4.bucket,
+      checksum_value: fakeFile4.checksum,
+      checksum_type: fakeFile4.checksumType,
+      key: fakeFile4.key,
+      path: null,
+      file_size: fakeFile4.size.toString(),
+      file_name: fakeFile4.fileName,
+      source: fakeFile4.source,
+      type: fakeFile4.type,
+    }
+  );
+
+  t.teardown(async () => {
+    await t.context.granulePgModel.delete(t.context.knex, { cumulus_id: records[0].cumulus_id });
+    // shouldn't need to delete files???
+  });
+});
+
+test.only('migrateGranuleAndFilesViaTransaction updates previously migrated file records correctly if the files are modified and the granule is re-migrated', async (t) => {
+  const {
+    knex,
+    testGranule,
+  } = t.context;
+
+  const testFile1 = testGranule.files[0];
+  const fakeFile2 = fakeFileFactory({
+    bucket,
+    key: cryptoRandomString({ length: 10 }),
+    size: 1098034,
+    fileName: cryptoRandomString({ length: 20 }),
+    checksum: 'checkSum02',
+    checksumType: 'md5',
+    type: 'data',
+    source: 'source2',
+  });
+  testGranule.files.push(fakeFile2);
+
+  // migrate 1st time
+  await migrateGranuleAndFilesViaTransaction({
+    dynamoRecord: testGranule,
+    knex,
+    loggingInterval: 1,
+  });
+
+  delete testFile1.size;
```

Comment:
suggest to use `modifiedTestFile1` `modifedFakeFile2`  etc. instead of update the original files. e.g.
const modifiedTestFile1 = omit(testFile1, ['size', ...])
const modifedFakeFile2 = {
...fakeFile2,
key: 'new key 2',
}

Reply:
updated